src/cryptography/hazmat/backends/openssl/decode_asn1.py

```diff
@@ -489,9 +489,15 @@ def _decode_extended_key_usage(backend, sk):
 
 def _decode_crl_distribution_points(backend, cdps):
     cdps = backend._ffi.cast("Cryptography_STACK_OF_DIST_POINT *", cdps)
-    cdps = backend._ffi.gc(cdps, backend._lib.sk_DIST_POINT_free)
-    num = backend._lib.sk_DIST_POINT_num(cdps)
 
+    dp_freefunc = backend._ffi.addressof(
+        backend._lib.original_lib, "DIST_POINT_free"
```

Comment:
I think this approach is fine, although maybe we should leave it as a private attribute?